src/nullablevector.jl

```diff
@@ -311,3 +311,14 @@ function Base.empty!(X::NullableVector)
     empty!(X.isnull)
     return X
 end
+
+notnullarray = Union{filter!(x -> !isa(x, Type{NullableArray}), subtypes(AbstractArray))...}
```

Comment:
I now realize that if I call `typed_hcat`, `typed_vcat` instead of calling `hcat/vcat` I can probably avoid the StackOverflow while using the simpler `AbstractArray` as you suggested.